### PR TITLE
Data api

### PR DESCRIPTION
**Pull request by DrPandemic**
Adds the `insert` with quorum. It uses the dialyzer PR. By playing with it, I found an issue in `validate_pk_uniqueness`. I'll do another PR to fix it.

### CODE REVIEW

File: lib/masdb/data_api.ex [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+         :ok <- validate_remote_insert(remote_answers) do
+      {:ok, local_row_id}
+    end
+
+  end
+  def insert(_, _, consistency, _) when consistency in [:unique, :multiple, :quorum, :all] do
+    :not_implemented
+  end
+  def insert(_, _, _, _) do
+    :unacceptable_consistency_level
```
Comment (samdus) [inline lib/masdb/data_api.ex]: I really like this, should we do this for Data.Map.select ?

File: test/data_api_test.exs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    c0 = %Schema.Column{is_pk: true,  name: "c1", type: :int}
+    s0 = %Schema{name: "foo", replication_factor: 0, columns: [c0]}
+    Register.Server.add_schema(s0, register)
+    this = self()
+    opts = [
+      timeout: 10,
+      fetch_fn: fn(_, _, _, _, _) -> send this, :fetched end
+    ]
+
+    insert("foo", %{"c1" => 0}, :unique, opts)
```
Comment (samdus) [inline test/data_api_test.exs]: Isn't it suppose to be ":not_implemented"

File: lib/masdb/data_api.ex (hunk 1 above)
Comment (DrPandemic) [inline lib/masdb/data_api.ex]: Yes

File: test/data_api_test.exs (hunk 2 above)
Comment (DrPandemic) [inline test/data_api_test.exs]: Well, yes. But it's also not doing a remote call... It was in prevision :P

**Review by samdus — APPROVED** [review]

**Pull request merged**